geopandas/io/tests/test_file.py

```diff
@@ -252,6 +258,22 @@ def test_to_file_int64(tmpdir, df_points):
     assert_geodataframe_equal(df_read, df, check_dtype=False, check_like=True)
 
 
+def test_overflow_int32_mapinfo(tmpdir, df_points):
```

Comment:
In terms of this issue with large ints, I think the best we can do is to detect this before writing tab files and instead output as a float64 column. `pd.to_numeric` can do downcasting to columns if possible, but is probably not quite suited to this situation.

Reply:
I guess we're kind o stuck here. I'll leave the failing test and look into testing the dtypes after read-in with fiona to see if we can make some assertions there.

Reply:
So pd.to_numeric_internally calls `pandas.core.dtypes.cast.maybe_downcast_numeric`, which is not part of the public api unfortunately (we could also factor part of this out to do this check ourselves, but I'd prefer not to have to maintain that check inside of geopandas. But we could do something like this within the mapinfo driver check:
```python
for c in world.columns[world.dtypes=="int64"]:
    # world[c] =1
    res = pd.to_numeric(world[c], downcast='integer')
    if res.dtype=="int64":
        warnings.warn(f"Int64 column '{c}' written to file as a float64 column as Mapinfo File does not support 64-bit integers")
    else:
        world.loc[:, c] = res
```

(it's not particularly efficient or elegant but it would allow you to move forward)